rbc/omnisci_backend/omnisci_table_function_manager.py

```diff
@@ -0,0 +1,102 @@
+__all__ = ['OmnisciTableFunctionManagerType']
+
+
+from numba.core import extending, types
+from numba.core.cgutils import make_bytearray, global_constant
+from rbc import structure_type, irutils
+from rbc.errors import UnsupportedError
+from rbc.targetinfo import TargetInfo
+from rbc.typesystem import Type
+from llvmlite import ir
+
+
+class OmnisciTableFunctionManagerType(Type):
+    """TableFunctionManager<> is a typesystem custom type that
+    represents a class type with the following public interface:
+
+      class TableFunctionManager
+      {
+        int32_t error_message(const char* message);
+        void set_output_row_size(int64_t num_rows);
+      }
+
+    """
+
+    @property
+    def __typesystem_type__(self):
+        ptr_t = Type.fromstring("int8 ptr")
+        return Type(ptr_t).params(NumbaPointerType=OmnisciTableFunctionManagerNumbaType).pointer()
+
+
+class OmnisciTableFunctionManagerNumbaType(structure_type.StructureNumbaPointerType):
+    pass
+
+
+error_msg = 'TableFunctionManager is only available in OmniSciDB 5.8 or newer'
+
+
+@extending.intrinsic
+def omnisci_udtfmanager_error_message_(typingctx, mgr, msg):
+    sig = types.int32(mgr, msg)
+
+    if not isinstance(msg, types.StringLiteral):
+        raise TypeError(f"expected StringLiteral but got {type(msg).__name__}")
+
+    def codegen(context, builder, signature, args):
+        mgr_ptr = args[0]
+
+        mgr_i8ptr = builder.bitcast(mgr_ptr, ir.IntType(8).as_pointer())
+
+        msg_bytes = msg.literal_value.encode('utf-8')
+        msg_const = make_bytearray(msg_bytes + b'\0')
+        msg_global_var = global_constant(
+            builder.module,
+            "table_function_manager_error_message",
+            msg_const)
+        msg_ptr = builder.bitcast(msg_global_var, ir.IntType(8).as_pointer())
+
+        fnty = ir.FunctionType(
+            ir.IntType(32), [
+                ir.IntType(8).as_pointer(), ir.IntType(8).as_pointer()])
```

Comment:
nit: you can declare the types in the global scope (`i8p = ir.IntType(8).as_pointer()`) and just use them here.
```suggestion
        fnty = ir.FunctionType(i32, [i8p, i8p])
```